### PR TITLE
Wireshark

### PR DESCRIPTION
**Pull request by taka-sasaki**
An update of Wireshark plugin for parsing the latest version of HopOpaqueField.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/194%23discussion_r33449315%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23discussion_r33449380%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23discussion_r33449659%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23discussion_r33449777%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23issuecomment-116592551%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23discussion_r33471908%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23issuecomment-116727428%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23issuecomment-121149205%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23issuecomment-121188703%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%2090022fcb6c6a06b3cca524ac057c91938efff702%20tools/wireshark/scion.lua%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23discussion_r33449315%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22maybe%20define%20%60CMN_HDR_LEN%3D8%60%20%3F%22%2C%20%22created_at%22%3A%20%222015-06-29T09%3A55%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22and%20IPV4_ADDR_LEN%3D4...%22%2C%20%22created_at%22%3A%20%222015-06-29T09%3A56%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/wireshark/scion.lua%3AL53-71%22%7D%2C%20%22Pull%2090022fcb6c6a06b3cca524ac057c91938efff702%20tools/wireshark/scion.lua%2088%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23discussion_r33449777%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22asd_length%20%3F%22%2C%20%22created_at%22%3A%20%222015-06-29T10%3A03%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22process_beacon%20function%20has%20not%20yet%20been%20updated.%5Cr%5CnI%20will%20update%20it%20later.%22%2C%20%22created_at%22%3A%20%222015-06-29T14%3A55%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9196745%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/taka-sasaki%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/wireshark/scion.lua%3AL216-235%22%7D%2C%20%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23issuecomment-116592551%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20think%20the%20rest%20is%20good.%5Cr%5Cnbtw.%20do%20you%20know%20any%20%28easy%29%20way%20to%20formalize%20our%20packet%20formats%20in%20some%20format/language%20that%20allows%20then%20for%20automagic%20translation%20to%20wireshark%20%3F%20Protocol%20Buffers%20%3F%22%2C%20%22created_at%22%3A%20%222015-06-29T10%3A09%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Unfortunately%2C%20I%20don%27t%20know%20such%20language.%5Cn%5CnI%20found%20a%20wireshark%20extension%20that%20provides%20APIs%20for%20writing%20plugin%20in%20Python.%20But%20it%20requires%20Python%202.7.%5Cn%5Cnhttps%3A//github.com/ashdnazg/pyreshark%5Cn%22%2C%20%22created_at%22%3A%20%222015-06-29T15%3A15%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9196745%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/taka-sasaki%22%7D%7D%2C%20%7B%22body%22%3A%20%22Is%20this%20still%20up%20to%20date%3F%20Can%20we%20get%20it%20merged%3F%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-07-14T07%3A08%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3840858%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shitz%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes.%20Please%20merge%20it.%22%2C%20%22created_at%22%3A%20%222015-07-14T09%3A58%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9196745%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/taka-sasaki%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2090022fcb6c6a06b3cca524ac057c91938efff702%20tools/wireshark/scion.lua%2031%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/194%23discussion_r33449659%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22isd_id%20is%20first%2012%20bits%2C%20ad_id%20is%20remaining%2020%20bits.%22%2C%20%22created_at%22%3A%20%222015-06-29T10%3A01%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/wireshark/scion.lua%3AL126-152%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 90022fcb6c6a06b3cca524ac057c91938efff702 tools/wireshark/scion.lua 7'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/194#discussion_r33449315'>File: tools/wireshark/scion.lua:L53-71</a></b>
- <a href='https://github.com/pszalach'></a> maybe define `CMN_HDR_LEN=8` ?
- <a href='https://github.com/pszalach'></a> and IPV4_ADDR_LEN=4...
- [x] <a href='#crh-comment-Pull 90022fcb6c6a06b3cca524ac057c91938efff702 tools/wireshark/scion.lua 31'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/194#discussion_r33449659'>File: tools/wireshark/scion.lua:L126-152</a></b>
- <a href='https://github.com/pszalach'></a> isd_id is first 12 bits, ad_id is remaining 20 bits.
- [x] <a href='#crh-comment-Pull 90022fcb6c6a06b3cca524ac057c91938efff702 tools/wireshark/scion.lua 88'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/194#discussion_r33449777'>File: tools/wireshark/scion.lua:L216-235</a></b>
- <a href='https://github.com/pszalach'></a> asd_length ?
- <a href='https://github.com/taka-sasaki'></a> process_beacon function has not yet been updated.
  I will update it later.
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/194#issuecomment-116592551'>General Comment</a></b>
- <a href='https://github.com/pszalach'></a> I think the rest is good.
  btw. do you know any (easy) way to formalize our packet formats in some format/language that allows then for automagic translation to wireshark ? Protocol Buffers ?
- <a href='https://github.com/taka-sasaki'></a> Unfortunately, I don't know such language.
  I found a wireshark extension that provides APIs for writing plugin in Python. But it requires Python 2.7.
  https://github.com/ashdnazg/pyreshark
- <a href='https://github.com/shitz'></a> Is this still up to date? Can we get it merged?
- <a href='https://github.com/taka-sasaki'></a> Yes. Please merge it.

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/194?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/194?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/194'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: tools/wireshark/scion.lua [hunk 1]
```diff
@@ -53,15 +53,18 @@ function scion_proto.dissector(buffer,pinfo,tree)
 	--debug
 	--srclen=4
 	--dstlen=4
-	
+	ISD_AD_LEN=4 
+
 	srcaddr=buffer(8,srclen)
```
Comment (pszal) [inline tools/wireshark/scion.lua]: maybe define `CMN_HDR_LEN=8` ?

Comment (pszal) [inline tools/wireshark/scion.lua]: and IPV4_ADDR_LEN=4...

File: tools/wireshark/scion.lua [hunk 2]
```diff
@@ -123,9 +126,26 @@ end
 function process_data(buffer,pinfo,tree)
 	--check src and destination address
 	local srcdst_field = ProtoField.string("srcdst","Source/Destination address")
-	local srcdst_tree = scion_tree:add(srcdst_field,buffer(8,8),"Source/Destination address")
-	srcdst_tree:add(buffer(8,srclen),"Source adress: " .. srcaddr)
-	srcdst_tree:add(buffer(8+srclen,dstlen),"Destination adress: " .. dstaddr)
+	local srcdst_tree = scion_tree:add(srcdst_field,buffer(8,srclen+dstlen),"Source/Destination address")
+
+	local srcaddr=buffer(8,srclen)
+	local src_isd_id=bit.rshift(buffer(8,4):uint(),20)  --first 20 bits are ISD ID
```
Comment (pszal) [inline tools/wireshark/scion.lua]: isd_id is first 12 bits, ad_id is remaining 20 bits.

File: tools/wireshark/scion.lua [hunk 3]
```diff
@@ -192,10 +216,19 @@ function process_beacon(buffer,pinfo,tree)
 		local signature_length= buffer(of_offset+8+4,2):uint()
 		local signature_length= buffer(of_offset+8+6,2):uint()
```
Comment (pszal) [inline tools/wireshark/scion.lua:231]: asd_length ?

Comment (taka-sasaki) [inline tools/wireshark/scion.lua:231]: process_beacon function has not yet been updated.
I will update it later.

**Pull request merged**